### PR TITLE
[youporn] Fixed the extractor by simplifying some parts

### PR DESCRIPTION
**Pull request by error39**

### CODE REVIEW

File: youtube_dl/extractor/youporn.py [hunk 1]
```diff
@@ -111,11 +93,7 @@ def _real_extract(self, url):
 
         return {
             'id': video_id,
-            'uploader': video_uploader,
-            'upload_date': upload_date,
             'title': video_title,
-            'thumbnail': thumbnail,
-            'description': video_description,
```
Comment (dstftw) [inline youtube_dl/extractor/youporn.py]: All this metadata is available and should be extracted.

**Comment by error39** [comment]
thanks, fixed it.

File: youtube_dl/extractor/youporn.py [hunk 2]
```diff
@@ -111,11 +96,9 @@ def _real_extract(self, url):
 
         return {
             'id': video_id,
-            'uploader': video_uploader,
-            'upload_date': upload_date,
```
Comment (dstftw) [inline youtube_dl/extractor/youporn.py]: These are also available.

File: youtube_dl/extractor/youporn.py [hunk 3]
```diff
@@ -45,35 +43,22 @@ def _real_extract(self, url):
...
-
         self.report_extraction(video_id)
         try:
-            video_title = params['title']
-            upload_date = unified_strdate(params['release_date_f'])
-            video_description = params['description']
-            video_uploader = params['submitted_by']
-            thumbnail = params['thumbnails'][0]['image']
+            video_title = self._html_search_regex(r'page_params.video_title = \'(.+?)\';', webpage, 'video URL')
+            video_description = self._html_search_regex(r'<meta name="description" content="(.+?)" />', webpage, 'video DESC')
```
Comment (yan12125) [inline youtube_dl/extractor/youporn.py]: ```self._html_search_meta``` is better here.

File: youtube_dl/extractor/youporn.py [hunk 4]
```diff
@@ -45,35 +43,22 @@ def _real_extract(self, url):
...
+            video_thumbnail = self._html_search_regex(r'<img src=\'(.+?)\' class="flipbook"', webpage, 'video THUMB')
         except KeyError:
```
Comment (yan12125) [inline youtube_dl/extractor/youporn.py]: ```_html_search_regex``` or ```_html_search_meta``` does not raise ```KeyError``` in case of not found.

File: youtube_dl/extractor/youporn.py (hunk 3 above)
Comment (error39) [inline youtube_dl/extractor/youporn.py]: thanks, changed it.

File: youtube_dl/extractor/youporn.py (hunk 2 above)
Comment (error39) [inline youtube_dl/extractor/youporn.py]: Added them

File: youtube_dl/extractor/youporn.py (hunk 4 above)
Comment (error39) [inline youtube_dl/extractor/youporn.py]: Could you suggest the right method? I'm a python and regex n00b btw :(

Comment (yan12125) [inline youtube_dl/extractor/youporn.py]: Just remove the ```try:``` and ```except:``` lines.

Comment (error39) [inline youtube_dl/extractor/youporn.py]: Thanks :)

**Comment by error39** [comment]
Not much seems to be happening now, is a new pull request necessary or should I just wait for someone to review & okay the submitted code?

Hope not to sound impatient, but I just don't know what the regular procedure for this is..

File: youtube_dl/extractor/youporn.py [hunk 5]
```diff
@@ -27,10 +27,11 @@ class YouPornIE(InfoExtractor):
...
             'id': '505835',
             'ext': 'mp4',
-            'upload_date': '20101221',
-            'description': 'Love & Sex Answers: http://bit.ly/DanAndJenn -- Is It Unhealthy To Masturbate Daily?',
-            'uploader': 'Ask Dan And Jennifer',
             'title': 'Sex Ed: Is It Safe To Masturbate Daily?',
+            'description': 'Watch Sex Ed: Is It Safe To Masturbate Daily? at YouPorn.com - YouPorn is the biggest free porn tube site on the net!',
+            'uploader': 'Ask Dan And Jennifer',
+            'thumbnail': 'http://cdn5.image.youporn.phncdn.com/201012/17/505835/640x480/8/sex-ed-is-it-safe-to-masturbate-daily-8.jpg',
+            'date': 'December 21, 2010',
```
Comment (dstftw) [inline youtube_dl/extractor/youporn.py]: There is no such field. Should be `upload_date` in `YYYYMMDD`.

File: youtube_dl/extractor/youporn.py [hunk 6]
```diff
@@ -45,35 +46,21 @@ def _real_extract(self, url):
...
-            video_description = params['description']
-            video_uploader = params['submitted_by']
-            thumbnail = params['thumbnails'][0]['image']
-        except KeyError:
-            raise ExtractorError('Missing JSON parameter: ' + sys.exc_info()[1])
+        video_title = self._html_search_regex(r'page_params.video_title = \'(.+?)\';', webpage, 'video URL')
+        video_description = self._html_search_meta('description', webpage, 'video DESC')
+        video_thumbnail = self._html_search_regex(r'page_params.imageurl\t=\t"(.+?)";', webpage, 'video THUMB')
+        video_uploader = self._html_search_regex(r"<div class=\'videoInfoBy\'>By:</div>\n<a href=\"[^>]+\">(.+?)</a>", webpage, 'video UPLOADER')
+        video_date = self._html_search_regex(r"<div class='videoInfoTime'>\n<i class='icon-clock'></i> (.+?)\n</div>", webpage, 'video DATE')
```
Comment (dstftw) [inline youtube_dl/extractor/youporn.py]: All these should be `fatal=False`.

File: youtube_dl/extractor/youporn.py [hunk 7]
```diff
@@ -45,35 +46,21 @@ def _real_extract(self, url):
...
-                                                webpage, 'download list').strip()
-        LINK_RE = r'<a href="([^"]+)">'
+        DOWNLOAD_LIST_RE = r'(?s)<div id="downloadModal" class="modalBox">(?P<testje>.*?)<div id="embedModal" class="modalBox">'
+        download_list_html = self._search_regex(DOWNLOAD_LIST_RE, webpage, 'testje').strip()
+        LINK_RE = r'<a href=\'([^"]+)\' title=\'Download Video\'>'
         links = re.findall(LINK_RE, download_list_html)
 
         # Get all encrypted links
-        encrypted_links = re.findall(r'var encryptedQuality[0-9]{3}URL = \'([a-zA-Z0-9+/]+={0,2})\';', webpage)
+        encrypted_links = re.findall(r'encryptedQuality[0-9]{3}URL\t=\t\'([a-zA-Z0-9+/]+={0,2})\';', webpage)
```
Comment (dstftw) [inline youtube_dl/extractor/youporn.py]: Use `\s*` instead.

File: youtube_dl/extractor/youporn.py [hunk 8]
```diff
@@ -45,35 +46,21 @@ def _real_extract(self, url):
...
 
         # Get all of the links from the page
-        DOWNLOAD_LIST_RE = r'(?s)<ul class="downloadList">(?P<download_list>.*?)</ul>'
-        download_list_html = self._search_regex(DOWNLOAD_LIST_RE,
-                                                webpage, 'download list').strip()
-        LINK_RE = r'<a href="([^"]+)">'
+        DOWNLOAD_LIST_RE = r'(?s)<div id="downloadModal" class="modalBox">(?P<testje>.*?)<div id="embedModal" class="modalBox">'
+        download_list_html = self._search_regex(DOWNLOAD_LIST_RE, webpage, 'testje').strip()
+        LINK_RE = r'<a href=\'([^"]+)\' title=\'Download Video\'>'
         links = re.findall(LINK_RE, download_list_html)
```
Comment (dstftw) [inline youtube_dl/extractor/youporn.py]: This extracts mess instead of download URL:
```
>youtube-dl http://www.youporn.com/watch/505835/sex-ed-is-it-safe-to-masturbate-daily/  -f 720p-1500k-cdn2b-0 -g
http://cdn2b.download.youporn.phncdn.com/201012/17/505835/720p_1500k_505835/YouPorn%20-%20Sex%20Ed%20Is%20It%20Safe%20To%20Masturbate%20Daily.mp4?rs=200&ri=2500
&s=1445177200&e=1445350800&h=178da8509b9b869e9ad12809e016f078' title='Download Video'>MP4 HD - For Windows 7/8, Mac and iPad</a>
<span class='downloadsize'>(37 MB)</span>
</div>
<div class='downloadOption'>
<i class='icon icon-download to-text-green'></i>
<a href='http://cdn2b.download.youporn.phncdn.com/201012/17/505835/480p_370k_505835/YouPorn%20-%20Sex%20Ed%20Is%20It%20Safe%20To%20Masturbate%20Daily.mp4?rs=150
&ri=1000&s=1445177200&e=1445350800&h=e5b5291f1ade3a2ea66af312daf37a04' title='Download Video'>MP4 - For Windows 7/8, Mac and iPad</a>
<span class='downloadsize'>(9 MB)</span>
</div>
<div class='downloadOption'>
<i class='icon icon-download to-text-green'></i>
<a href='http://cdn2b.download.youporn.phncdn.com/201012/17/505835/240p_240k_505835/YouPorn%20-%20Sex%20Ed%20Is%20It%20Safe%20To%20Masturbate%20Daily.mp4?rs=150
&ri=1000&s=1445177200&e=1445350800&h=7b1ceb68fd3affe8f2308a6501cbb218' title='Download Video'>MP4 - For iPhone/iPod</a>
<span class='downloadsize'>(6 MB)</span>
</div>
<div class='downloadOption'>
<i class='icon icon-download to-text-green'></i>
<a href='http://cdn2b.download.youporn.phncdn.com/201012/17/505835/180p_150k_505835/YouPorn%20-%20Sex%20Ed%20Is%20It%20Safe%20To%20Masturbate%20Daily.3gp?rs=150
&ri=1000&s=1445177200&e=1445350800&h=c8ce8aa7f320c8c9253a1a3a6817539c
```

**Comment by error39** [comment]
Thanks for the feedback Sergey, fixed the issues and also learned a bit :)

File: youtube_dl/extractor/youporn.py [hunk 9]
```diff
@@ -45,45 +47,23 @@ def _real_extract(self, url):
...
+        DOWNLOAD_LIST_RE = r'(?s)sources: {\n(?P<testje>.*?)}'
+        download_list_html = self._search_regex(DOWNLOAD_LIST_RE, webpage, 'testje').strip()
+        LINK_RE = r': \'(.+?)\','
         links = re.findall(LINK_RE, download_list_html)
 
-        # Get all encrypted links
-        encrypted_links = re.findall(r'var encryptedQuality[0-9]{3}URL = \'([a-zA-Z0-9+/]+={0,2})\';', webpage)
-        for encrypted_link in encrypted_links:
-            link = aes_decrypt_text(encrypted_link, video_title, 32).decode('utf-8')
-            links.append(link)
```
Comment (dstftw) [inline youtube_dl/extractor/youporn.py]: Why did you remove this?

File: youtube_dl/extractor/youporn.py [hunk 10]
```diff
@@ -45,45 +47,23 @@ def _real_extract(self, url):
...
-
         formats = []
         for link in links:
-            # A link looks like this:
-            # http://cdn1.download.youporn.phncdn.com/201210/31/8004515/480p_370k_8004515/YouPorn%20-%20Nubile%20Films%20The%20Pillow%20Fight.mp4?nvb=20121113051249&nva=20121114051249&ir=1200&sr=1200&hash=014b882080310e95fb6a0
-            # A path looks like this:
-            # /201210/31/8004515/480p_370k_8004515/YouPorn%20-%20Nubile%20Films%20The%20Pillow%20Fight.mp4
```
Comment (dstftw) [inline youtube_dl/extractor/youporn.py]: And this?

File: youtube_dl/extractor/youporn.py (hunk 9 above)
Comment (error39) [inline youtube_dl/extractor/youporn.py]: After changing the regex to `page_params.encryptedQuality[0-9]{3,4}URL\s=\s\'([a-zA-Z0-9+/]+={0,2})\';` it appends the same url's which were already gathered in the links array again, so this  code seemes redundant to me, or isn't it?

Comment (dstftw) [inline youtube_dl/extractor/youporn.py]: AFAIR, previously some hd videos were only available via encrypted URL. Have you checked this now?

Comment (error39) [inline youtube_dl/extractor/youporn.py]: I didn't know that, so I tested roughly a dozen random 720p (their maximum quality) videos from the HD category but it returned no 'extra' urls in the encrypted links.
We could leave it in and remove duplicate url's in the final links array or consider it outdated and remove this part all together, what do you think?

File: youtube_dl/extractor/youporn.py (hunk 10 above)
Comment (error39) [inline youtube_dl/extractor/youporn.py]: I can't remember why, but I've put it back in now with the corresponding example url from the _TEST array, I'll commit now..

**Comment by error39** [comment]
Is this really the desired way of 'communicating' with people trying to help out?
Thank you for your efforts in fixing it anyway, it works much better now, I suppose...

**Comment by dstftw** [comment]
I've not yet had time to post here, sorry. I'm still working on it.
I've cherry-picked you changes, thanks.

**Comment by error39** [comment]
Whoops, sorry.. my misunderstanding..
Thanks for picking it up yourself, anything more I can do?

Op 25/10/15 om 18:35 schreef Sergey M.:
